zilliqa/src/state.rs

```diff
@@ -45,7 +45,7 @@ impl State {
         let _ = state.set_gas_price(default_gas_price().into());
 
         for (address, balance) in genesis_accounts {
-            state.set_native_balance(*address, (*balance).into())?;
```

Comment:
I'm not very familiar with TF so I'm sure there must've been a good reason to change this to a `String`, but this seems like the kind of thing I'd love to validate early rather than late. Maybe even with a custom serde deserializer function specified in cfg.rs, to validate it on parse? Actually, I'm not sure off the top of my head, but can it be specified as a string but deserialized directly into an `u128`?

I'm just not super keen on having a string that can fail parsing at any time be stored in the config.

But also pragmatically speaking we are almost certainly going to construct the `State` very soon after booting the node, so in practical terms it will still fail quite fast, so only worth looking into if you think you can add the extra type-safety without too much effort.